### PR TITLE
Qt sidebar

### PR DESCRIPTION
**Pull request by ksoze95**
Resolves: #4454

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: src/qt/EditToolbarHandler.h [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #ifndef EDIT_TOOLBAR_HANDLER_H
-#define EDIT_TOOLBAR_HANDELR_H
+#define EDIT_TOOLBAR_HANDLER_H
```
Comment (crsib) [inline src/qt/EditToolbarHandler.h:2]: TBH I prefer `#pragma once` exactly for this reason

File: src/qt/uicomponents/MeterPanelHandler.cpp [hunk 2]
```diff
@@ -13,15 +13,7 @@ qreal MeterPanelHandler::Value() const
...
-   if (value < 0.0) {
-      qDebug() << "MeterPanelHandler value cannot be less than 0";
-      return;
-   }
-
-   if (value > 1.0) {
-      qDebug() << "MeterPanelHandler value cannot be greater than 1";
-      return;
-   }
+   Q_ASSERT(value >= 0.0 && value <= 1.0);
```
Comment (crsib) [inline src/qt/uicomponents/MeterPanelHandler.cpp:16]: 1. I would prefer the standard `assert`
2. Asserting should not replace the checks

File: src/qt/qml/main.qml [hunk 3]
```diff
@@ -93,9 +93,17 @@ ApplicationWindow {
       id: toolsToolbar
```
Comment (crsib) [inline src/qt/qml/main.qml:93]: I dunno why we need this, but no objection for now

File: src/qt/EditToolbarHandler.h (hunk 1 above)
Comment (ksoze95) [reply src/qt/EditToolbarHandler.h]: Replaced with pragma.

File: src/qt/uicomponents/MeterPanelHandler.cpp (hunk 2 above)
Comment (ksoze95) [reply src/qt/uicomponents/MeterPanelHandler.cpp]: Done.

File: src/qt/uicomponents/TrackType.h [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#ifndef TRACK_TYPE_H
+#define TRACK_TYPE_H
+
+#include <QtCore/QtCore>
+#include <QtQml/qqmlregistration.h>
+
+class TrackType
```
Comment (crsib) [inline src/qt/uicomponents/TrackType.h:7]: We will need to figure out how to make it open ended. No assumption should be made about the available track types...

**Review by crsib — APPROVED** [review]
I wonder if this should be squash merged

**Pull request merged**